adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -1,47 +1,27 @@
 package caliban
 
-import caliban.execution.QueryExecution
 import caliban.interop.tapir.ws.Protocol
-import caliban.interop.tapir.{ RequestInterceptor, TapirAdapter, WebSocketHooks }
+import caliban.interop.tapir.{ HttpAdapter, WebSocketHooks }
 import sttp.tapir.Codec.JsonCodec
 import sttp.tapir.DecodeResult
 import sttp.tapir.server.ziohttp.{ ZioHttpInterpreter, ZioHttpServerOptions }
-import zio.http._
+import zio._
 import zio.http.ChannelEvent.UserEvent.HandshakeComplete
 import zio.http.ChannelEvent.{ ChannelRead, UserEventTriggered }
+import zio.http._
 import zio.http.socket._
-import zio._
 import zio.stream._
 
 object ZHttpAdapter {
 
-  def makeHttpService[R, E](
-    interpreter: GraphQLInterpreter[R, E],
-    skipValidation: Boolean = false,
-    enableIntrospection: Boolean = true,
-    queryExecution: QueryExecution = QueryExecution.Parallel,
-    requestInterceptor: RequestInterceptor[R] = RequestInterceptor.empty
-  )(implicit
-    requestCodec: JsonCodec[GraphQLRequest],
-    responseCodec: JsonCodec[GraphQLResponse[E]],
+  def makeHttpService[R, E](adapter: HttpAdapter[R, E])(implicit
     serverOptions: ZioHttpServerOptions[R] = ZioHttpServerOptions.default[R]
-  ): HttpApp[R, Throwable] = {
-    val endpoints = TapirAdapter.makeHttpService[R, E](
-      interpreter,
-      skipValidation,
-      enableIntrospection,
-      queryExecution,
-      requestInterceptor
-    )
-    ZioHttpInterpreter(serverOptions).toHttp(endpoints)
-  }
+  ): HttpApp[R, Throwable] =
+    ZioHttpInterpreter(serverOptions).toHttp(adapter.serverEndpoints[R])
 
   def makeWebSocketService[R, E](
     interpreter: GraphQLInterpreter[R, E],
-    skipValidation: Boolean = false,
-    enableIntrospection: Boolean = true,
     keepAliveTime: Option[Duration] = None,
-    queryExecution: QueryExecution = QueryExecution.Parallel,
     webSocketHooks: WebSocketHooks[R, E] = WebSocketHooks.empty
```

Comment:
Could we add `WebSocketHooks` and `keepAliveTime` to the configurator too? Or would that be confusing you think?

Reply:
I thought about it, but I kinda like to have them only for WebSockets, I don't really want to put them in `ExecutionConfiguration`.
It could be a separate case class and fiber ref, but there's also the problem that `WebSocketHooks` is parameterized in R and E.